polyglot/piranha/test-resources/java/insert_field_and_initializer/configurations/rules.toml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2022 Uber Technologies, Inc.
+# 
+# <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License at
+# <p>http://www.apache.org/licenses/LICENSE-2.0
+# 
+# <p>Unless required by applicable law or agreed to in writing, software distributed under the
+# License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains rules to the specific feature flag API.
+
+#
+# For @stale_flag_name = STALE_FLAG and @treated = true
+# Before 
+#  exp.isToggleEnabled(Experiment.STALE_FLAG)
+# After 
+#  true
+#
+[[rules]]
+name = "add_field_declaration"
+query = """(
+(class_declaration name: (_)@class_name 
+    body : (class_body ((_)*) @class_members)  @class_body
+ ) @class_declaration
+(#eq? @class_name "FooBar")
+)"""
+replace_node = "class_body"
+replace = "{\n private List<String> names;\n @class_members \n}"
+[[rules.constraints]]
+matcher = "(class_declaration ) @c_cd"
+queries = ["""(
+(field_declaration (variable_declarator name:(_) @name )) @field
+(#eq? @name "names")
+)"""]
+
+[[rules]]
+name = "add_import"
+query = """
+(package_declaration) @pkg_dcl
+"""
+replace_node = "pkg_dcl"
+replace = "@pkg_dcl \n import java.util.List;"
+holes = ["class_name"]
```

Comment:
Ok. 
# Holes = ["class_name"]
`holes= ["class_name"]` is legacy, I forgot I had the `groups=["Cleanup Rule"]` feature in `master` (my bad). I will update the code. I had put it as a precondition to ensure that `add_import` does not get triggered before `add_field_declaration`.  I will fix this. Now that we can add  `groups=["Cleanup Rule"]` to a rule, this triggering problem will not happen. 


# Explanation of the test case: 
First the code matches `add_field_declaration`, and adds a field declaration. 
Note that when it rewrites the code it will change the class:
```
package com.uber.piranha;
import java.util.ArrayList;

class FooBar{

  private String address;

  private String phoneNumber;

  private String prefix;

  private String street;

  public String getAddress() {
    return address;
  } 

  public String getPhoneNumber() {
    return phoneNumber;
  }

  public String getStreet() {
    return street;
  }

  public String getPrefix() {
    return prefix;
  }

  public FooBar(String address) {
    this.address = address;
  }

}
```
to 
```
package com.uber.piranha;
import java.util.ArrayList;

class FooBar{
  private List<String> names;
  private String address;
  private String phoneNumber;
  private String prefix;
  private String street;
  public String getAddress() {
    return address;
  } 
  public String getPhoneNumber() {
    return phoneNumber;
  }
  public String getStreet() {
    return street;
  }
  public String getPrefix() {
    return prefix;
  }
  public FooBar(String address) {
    this.address = address;
  }
}
```

Note here the new lines between getters are removed.
That causes the matched range to be larger than replace range :| Thus causing an exception. 
We are okay with this happening, because we run `google-java-format` / `swift-lint` which handles this problem mess before creating the PR.

Reply:
Ah, ok, I missed the whitespace changes. Are those due to tree-sitter itself or part of polyglot Piranha rules? I guess either way they happen specifically because the `add_field_declaration` rule technically rewrites the whole class, correct?